### PR TITLE
Add workflow diagram to pull request doc

### LINKED ISSUE #13935: RFE: Review Workflow Diagram
Use Case: 

Developer has some cool new feature that they wish to add to kubernetes. This diagram walks them through the process that the developer will follow to have their PR-reviewed in a timely fashion.  It will also set the expectations for all parties involved in order to remove ambiguity. 

@bgrant0607 
/cc @childsb  @kubernetes/rh-cluster-infra

### PR DESCRIPTION
**Pull request by j3ffml**
Fixes #13935

### CODE REVIEW

File: docs/devel/pull-requests.md [hunk 1]
```diff
@@ -34,8 +34,7 @@ Documentation for other releases can be found at
 Pull Request Process
 ====================
 
-An overview of how we will manage old or out-of-date pull requests.
-
+An overview of how we will manage old or out-of-date pull requests.k
```
Comment (ihmccreery) [inline docs/devel/pull-requests.md]: Remove "k" at end.  Oh Vim...

Comment (ihmccreery) [inline docs/devel/pull-requests.md]: Also, this line seems wrong.

Comment (jlowdermilk) [inline docs/devel/pull-requests.md]: removed the line, moved the section about old prs lower.

**Comment by ihmccreery** [comment]
This doc needs a lot more love than just the diagram.  Do you plan more than just the diagram?

Re: the diagram:

1. It'd be better if it flowed vertically so I didn't have to squint to see it.
1. It needs to take the review-edit-review-edit cycle.
1. Per above, it needs a lot more explanation than just what you can fit in boxes.

**Comment by jlowdermilk** [comment]
@ihmccreery PTAL. I updated the doc to explain process a bit better, and made diagram larger, vertical.

**Comment by jlowdermilk** [comment]
cc @sarahnovotny

**Comment by sarahnovotny** [comment]
looks good to talk about tomorrow.  (vertical was a good idea @ihmccreery )

File: docs/devel/pull-requests.md [hunk 2]
```diff
@@ -58,10 +48,40 @@ Either the [on call](https://github.com/kubernetes/kubernetes/wiki/Kubernetes-on
...
 
 Additionally, for infrequent or new contributors, we require the on call to apply the "ok-to-merge" label manually.  This is gated by the [whitelist](https://github.com/kubernetes/contrib/blob/master/mungegithub/whitelist.txt).
 
+### Before sending a pull request
+
+The following will save time for both you and your reviewer:
+
+* Enable [pre-commit hooks](development.md#committing-changes-to-your-fork) and verify they pass.
+* Verify `hack/verify-generated-docs.sh` passes.
+* Verify `hack/test-go.sh` passes.
```
Comment (ixdy) [inline docs/devel/pull-requests.md]: maybe also mention "Verify `make` passes"? There are a lot of PRs which don't build.

File: docs/devel/pull-requests.md [hunk 3]
```diff
@@ -58,10 +48,40 @@ Either the [on call](https://github.com/kubernetes/kubernetes/wiki/Kubernetes-on
...
+
+### Visual overview
+
+![PR workflow](pr_workflow.png)
+
+Other notes
```
Comment (ixdy) [inline docs/devel/pull-requests.md]: maybe make each of the two paragraphs here linkable sections so that you can link to them directly when closing PRs?

Comment (jlowdermilk) [inline docs/devel/pull-requests.md]: Done.

File: docs/devel/pull-requests.md (hunk 2 above)
Comment (jlowdermilk) [inline docs/devel/pull-requests.md]: Done.

**Comment by jlowdermilk** [comment]
Comments addressed. @ixdy or @ihmccreery ptal

**Comment by ihmccreery** [comment]
Thanks @jlowdermilk

**Pull request merged**